tsdb/engine/tsm1/digest.go

```diff
@@ -142,6 +142,7 @@ func DigestFresh(dir string, files []string, shardLastMod time.Time) (bool, stri
 	if err != nil {
 		return false, fmt.Sprintf("Can't read digest: err=%s", err)
 	}
+	defer dr.Close()
```

Comment:
`defer` is good to ensure cleanup on early return, but can you also check the error returned by `dr`, perhaps just after the last usage? `Close` should be idempotent.

Reply:
Good idea. Since this is a backport of something on `master`, I'll make a note and change it on a future commit.